### PR TITLE
feat(matter): adds new Matter Occupancy Sensor endpoint

### PR DESCRIPTION
**Pull request by SuGlider**
## Description of Change
Adds a new Matter Endpoint for Occupancy Sensor.

## Tests scenarios
ESP32-C6 using the provided example.

## Related links
Related to #7432

### CODE REVIEW

File: libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+const char *password = "your-password";  // Change this to your WiFi password
+
+// Button control
+uint32_t button_time_stamp = 0;                // debouncing control
+bool button_state = false;                     // false = released | true = pressed
+const uint32_t decommissioningTimeout = 5000;  // keep the button pressed for 5s, or longer, to decommission
+
+void setup() {
+  // Initialize the USER BUTTON (Boot button) that will be used to decommission the Matter Node
+  // The button will also be used to manually toggle the Occupancy Sensor state
```
Comment (me-no-dev) [inline libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino:51]: seems that the button is not used to toggle occupancy

File: libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  // Manually connect to WiFi
+  WiFi.begin(ssid, password);
+  // Wait for connection
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    Serial.print(".");
+  }
+  Serial.println();
+
+  // set initial occupancy sensor state as false and connected to a PIR sensor type (default)
```
Comment (me-no-dev) [inline libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino:65]: Sketch suggests that is not actually PIR that is being used (PIR produces short pulses when it detects movement), but rather mmWave type sensor that detects body and pulse is long (as long as the body is being detected in range). I base this on the fact that you simulate 2minutes long pulses

Comment (me-no-dev) [reply libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino:65]: I also wonder how would an ultrasonic sensor be implemented, given that it produces a range value and not a boolean pulse.

Comment (SuGlider) [reply libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino:65]: Under Matter perspective, it doesn't matter.
The sketch shall deal with whatever occupancy technology is used.
All Matter cares is about a boolean state :: Occupied or Vacant.

PIR / Ultrasonic is just an information that may or nor be displayed by the APP.

File: libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino (hunk 1 above)
Comment (SuGlider) [reply libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino]: Done - this was a left over of testing I was doing.

File: libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino (hunk 2 above)
Comment (SuGlider) [reply libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino:64]: > I also wonder how would an ultrasonic sensor be implemented, given that it produces a range value and not a boolean pulse.

I tested it with 10 seconds and a button used to toggle the Matter Boolean Occupancy flag.
Matter doesn't deal with any working Sensor functionality. It just cares about the Boolean Flag.

Comment (me-no-dev) [reply libraries/Matter/examples/MatterOccupancySensor/MatterOccupancySensor.ino:64]: ok

**Review by me-no-dev — APPROVED** [review]

**Review by lucasssvaz — APPROVED** [review]

**Pull request merged**